src/endpoint/s3/s3_rest.js

```diff
@@ -286,8 +287,8 @@ async function authorize_request_policy(req) {
     if (!s3_policy) {
         // in case we do not have bucket policy
         // we allow IAM account to access a bucket that is owned by their root account
-        const is_iam_account_and_same_root_account_owner = account.owner !== undefined &&
-            owner_account && account.owner === owner_account.id;
+        const owner_id_to_compare = req.object_sdk.nsfs_config_root ? owner_account?.id : bucket_owner_id;
```

Comment:
I think its better to move this `req.object_sdk.nsfs_config_root` to a variable like `is_nsfs_nc`, I have seen repeated use of this, and will improve the readbility.